sim/battle.js

```diff
@@ -1795,7 +1793,18 @@ class Battle extends Dex.ModdedDex {
 		defense = this.runEvent('Modify' + statTable[defenseStat], defender, attacker, move, defense);
 
 		//int(int(int(2 * L / 5 + 2) * A * P / D) / 50);
-		let baseDamage = Math.floor(Math.floor(Math.floor(2 * level / 5 + 2) * basePower * attack / defense) / 50) + 2;
+		let baseDamage = Math.floor(Math.floor(Math.floor(2 * level / 5 + 2) * basePower * attack / defense) / 50);
+		
+		// Calculate initial damage modifiers (order differs between generations)
+		baseDamage = this.modifyDamage(pokemon, target, move, basePower, baseDamage, suppressMessages);
```

Comment:
Apparently, `basePower` is passed because it can be zero sometimes. In what situations is this true?